packages/hub/node-tests/routes/cards/delete-test.ts

```diff
@@ -66,5 +64,19 @@ if (process.env.COMPILER) {
 
       expect(existsSync(join(getRealmDir(), 'post0')), 'card is deleted from realm').to.be.false;
     });
+
+    it('can delete an existing card that has children', async function () {
+      await getCard(`${realmURL}post`).expect(200);
+      await deleteCard(`${realmURL}post`).expect(204);
+      await getCard(`${realmURL}post`).expect(404); // Deleted card is deleted!
+      await getCard(`${realmURL}post0`).expect(422); // card is in an error state
```

Comment:
yeah this was what i was mentioning while we were pairing--it seems weird that a card that is not under your control can invalidate your card, especially if it came from a different realm, given when the card being removed is included in your compiled card. this seems worth a discussion with @ef4 for how we want the system to behave.

like with the card updates, there might be an opt-in/out policy on accepting an upstream change based on the changed card being in your realm or not, or if you are in "dev" mode or not, or some other heuristic

Reply:
Fixing this will probably go with the work to do finer-grained invalidation & recompilation.

Each CompiledCard has within it *all* the details of its dependencies, which means you can in theory recompile just fine even if your parent or one of your fields has been deleted. It comes down to deciding which parts of the card needs to be recompiled and which should be preserved from the previous iteration.

Reply:
Agreed. Definitely needs to be discussed.

Reply:
@habdelra We decide to keep this aggressive for now. The scenario where we want the durable index can be added onto this when we flesh out what those realm contracts should look like. Right now, we're always just in dev, so we want to be "noisy." But once we have realms on different machines where you don't have control, that's when we'll want the durability.